lib/collection/src/collection_manager/segments_updater.rs

```diff
@@ -574,6 +597,49 @@ where
     Ok(res)
 }
 
+fn select_excluded_by_filter_ids(
+    segments: &SegmentHolder,
+    point_ids: impl IntoIterator<Item = PointIdType>,
+    filter: Filter,
+    hw_counter: &HardwareCounterCell,
+) -> CollectionResult<AHashSet<PointIdType>> {
+    // Filter for points that doesn't match the condition, and have matching
+    let non_match_filter = segment::types::Filter {
+        should: None,
+        min_should: None,
+        must: Some(vec![Condition::HasId(point_ids.into_iter().collect())]),
+        must_not: Some(vec![Condition::Filter(filter)]),
+    };
```

Comment:
We can use our new helper here:


```suggestion
    let non_match_filter =
        Filter::new_must_not(Condition::Filter(filter)).merge_with_ids(point_ids);
```